src/base/bittorrent/session.cpp

```diff
@@ -321,6 +321,11 @@ QString Session::tempPath() const
     return m_tempPath;
 }
 
+QString Session::torrentTempPath(const InfoHash &hash) const
+{
+    return m_tempPath + QString(hash) + "/";
```

Comment:
Just to do add: I think user is pretty safe with the truncated hash. I can't do the maths but I think for a researcher to find a valid info hash from our partial we would need to know all the torrent hashes in existance. Can he really do that? And even if he does the rest of the (40-7) chars can lead to other torrents too.